src/bugreporter/test/odemis_bugreporter_test.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Created on 17 October 2018
+
+@author: Philip Winkler
+
+Copyright © 2018 Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+
+"""
+
+import unittest
+import os
+import logging
+import zipfile
+from bugreporter import OdemisBugreporter
+from urllib2 import HTTPError
+
+REPORTER_TEST_ID = 12
+# Set TEST_SUPPORT_TICKET=1 to skip test cases which create a ticket, so we don't fill up
+# the osticket database with test messages.
+TEST_SUPPORT_TICKET = (os.environ.get("TEST_SUPPORT_TICKET", 0) != 0)
+TEST_SUPPORT_TEAM_EMAIL = "winkler@delmic.com"
+
+
+class TestOdemisBugreporter(unittest.TestCase):
+
+    def setUp(self):
+        self.bugreporter = OdemisBugreporter()
+
+    def test_search_api_key(self):
+        """
+        Tests search_api_key function.
+        """
+        # If test computer has key saved in .local/share/odemis or /usr/share/odemis,
+        # check this key, otherwise expect LookupError
+        customer_key_path = os.path.join(os.path.expanduser(u"~"), '.local', 'share',
+                                         'odemis', 'osticket.key')
+        fallback_key_path = os.path.join('usr', 'share', 'odemis', 'osticket.key')
+        if not os.path.isdir(customer_key_path) and not os.path.isdir(fallback_key_path):
+            self.assertRaises(LookupError, self.bugreporter.search_api_key)
+        else:
+            try:
+                api_key = self.bugreporter.search_api_key
+                self.assertEqual(type(api_key), str, "API key needs to be a string.")
+                self.assertFalse('\n' in api_key, "API key should not contain newline character")
+            except Exception as e:
+                self.fail("search_api_key failed with exception %s " % e)
+
+    def test_source_api_key(self):
+        """
+        Tests format of source api key.
+        """
+        key_path = (os.path.expanduser(u"~") +
```

Comment:
For that path, I usually prefer to hard-code it less. You can use:
`os.path.dirname(odemis.__file__)+ "../install/linux"`